contracts/abstracted_account.algo.ts

```diff
@@ -44,7 +58,7 @@ export class AbstractedAccount extends Contract {
   private verifyRekeyToAbstractedAccount(): void {
     let rekeyedBack = false;
 
-    for (let i = this.txn.groupIndex; i < this.txnGroup.length; i += 1) {
+    for (let i = (this.txn.groupIndex + 1); i < this.txnGroup.length; i += 1) {
```

Comment:
also incremented the starting index here because we dont need to do checks on the app call that initiates the rekey  to plugin